Removed pluralisation and updated session interface

morses-code opened the pull request:
Removed pluralisation of sessions where it wasn't supposed to happen
Updated the sessions interface, and sessions.go to match interface and noop implementations.

Just sanity check on changes.

morses-code committed:
Removed unintended pluralisation of session 

- Also made session.go more inline with the interface and noop implementation.

morses-code committed:
Removed redundant code

- `sess.New()` returns a session so the block below is unnecessary

morses-code committed:
Merge branch 'feature/add-deleteall-sessions-handler' into feature/remove-sessions-pluralisation

jessjenkins commented on api/interfaces.go:
```diff
@@ -15,8 +15,8 @@ type AuthHandler interface {
-// Sessions interface for getting a new session
-type Sessions interface {
+// Session interface for getting a new session
+type Session interface {
```
It is slightly confusing to see both the interface and struct `New` returns having the same name. By convention interfaces should end `er`.  but I'm not sure what exactIy I would replace it with. Maybe something like `SessionManager` or `SessionInitialiser`? I'm not brilliant at naming things though and most words I can come up with are ending in `…or`

morses-code replied:
No idea what to do with this one, I was reading as well that the convention is that one method interfaces are the name of the method with -er suffix. But I'm not sure Newer is approptiate... I'll look to see if I can give the method a better name. Or if its not really that big of a deal `SessionManager` feels a good fit.

jessjenkins commented on api/handlers.go:
```diff
@@ -15,7 +15,7 @@ import (
 
 // CreateSessionHandlerFunc returns a function that generates a session. Method = "POST"
-func CreateSessionHandlerFunc(sessions Sessions, cache Cache) http.HandlerFunc {
+func CreateSessionHandlerFunc(sess Session, cache Cache) http.HandlerFunc {
```
I don't personally like partial variable names like `sess`, I can see sense in short vars like `s` if the code is trivial enough or the variable is used frequently but personally I would expand this one to the full word. I'm not sure what our standards are on this though.

morses-code replied:
Having a bit of a dilema with this one, struggling to appropriately name the variables, will look at this though. I do agree.

morses-code committed:
Renaming variables and interfaces

- Updated variable names so they are more clear in purpose
- Updated "Session" interface, method is now called Update, renamed interface to Updater (in line with naming standards for one method interfaces)
- Updated mock to use new interface 
- Updated noop with new method name
- Updated session with new method name

jessjenkins commented on api/handlers.go:
```diff
@@ -15,7 +15,7 @@ import (
 
 // CreateSessionHandlerFunc returns a function that generates a session. Method = "POST"
-func CreateSessionHandlerFunc(sess Session, cache Cache) http.HandlerFunc {
+func CreateSessionHandlerFunc(updater Updater, cache Cache) http.HandlerFunc {
```
This is clearer that it is an interface now, but it's lost the context that it relates to a session. Can I suggest…
```suggestion
func CreateSessionHandlerFunc(session SessionUpdater, cache Cache) http.HandlerFunc {
```
Along with all the other changes to make this possible.

morses-code committed:
Updated Updater interface name

- Name is now less vague and a bit more obvious to its purpose
